### PR TITLE
node tag and label feature

### PR DESCRIPTION
**Pull request by lhw4d4**
@yyscamper @hohene @gpaulos

### CODE REVIEW

**Comment by rackhdmirror** [comment]
@RackHD/corecommitters please review.

File: test/fit_tests/config/stack100.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "bmc": "stack100-bmc.admin",
+    "control": "stack100-control.admin",
+    "data": "stack100-data.admin",
+    "hyper": "stack100-esxi.admin",
+    "ora": "stack100-ora.admin",
+    "ovamac": "00:50:56:00:11:00",
+    "pdu": "192.168.1.254",
+    "type": "esxi",
+    "nodes": [
+      {"bmcmac": "00:1e:67:d5:e1:96", "nodeid": "", "sku": "Rinjin KP"},
```
Comment (anhou) [inline test/fit_tests/config/stack100.json]: As I know, the server name listed in https://eos2git.cec.lab.emc.com/OnRack/emc-hardware-support could be published to public, there is a public  name such as 'platform-r' for Rxxjxx, so please confirm with Rod or someone for this

File: test/fit_tests/config/stack_config.json [hunk 2]
```diff
@@ -9,10 +9,10 @@
     "pdu": "PDU admin address",
     "type": "deployment type: esxi, docker, linux, or vagrant"
   },
-  "vagrant":{
-    "ora": "localhost",
-    "type": "vagrant"
-  },
+  "vagrant":{
```
Comment (anhou) [inline test/fit_tests/config/stack_config.json]: no change, but some extra spaces in the end

File: test/fit_tests/tests/rackhd20/test_rackhd20_api_tagmanagement.py [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+        return response_tag['text']
+    else:
+        return response_tag['status']
+
+def get_tag_list():
+    mon_url = '/api/{}/tags'.format(MON_API_VERSION)
+    response_tag = fit_common.rackhdapi(mon_url)
+    return response_tag['text']
+
+def get_tag_list_on_node(nodeid):
```
Comment (anhou) [inline test/fit_tests/tests/rackhd20/test_rackhd20_api_tagmanagement.py]: is there any plan to use swagger client pyton library?

File: test/fit_tests/tests/rackhd20/test_rackhd20_api_tagmanagement.py [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+        for tag in tag_list:
+            print "{} has tags: {}".format(node,tag)
+            response = delete_tag_in_node(node,tag)
+            if response  not in range(200,205):
+                print "Fail to delete the tag:" + tag
+                return 2
+    print "clear the environment finished!"
+    return 0
+
+def get_node_list_by_sku(skuname):
```
Comment (yyscamper) [inline test/fit_tests/tests/rackhd20/test_rackhd20_api_tagmanagement.py]: actually there is an API to get all nodes that belongs to the sku:
```
GET /api/2.0/skus/<skuId>/nodes
```

File: test/fit_tests/tests/rackhd20/test_rackhd20_api_tagmanagement.py (hunk 3 above)
Comment (lhw4d4) [inline test/fit_tests/tests/rackhd20/test_rackhd20_api_tagmanagement.py]: It seems that the swagger client python library is not in fit_test folder. If only using FIT to run tests, then the library cannot be imported. So I did not use the library.

Comment (hohene) [inline test/fit_tests/tests/rackhd20/test_rackhd20_api_tagmanagement.py]: There is a plan to include the swagger client once CIT/FIT get merged into one tool.  Then we'll have that library along with the restful calls available to use.

**Comment by johren** [comment]
@lhw4d4,  I'm checking in on the status of some of the older PRs.  It looks like this PR has been overcome by changes to the test infrastructure since this was submitted.  Would you like to continue forward and resolve the conflicts or would you like to close this PR at this point?

**Comment by johren** [comment]
To clean up older PRs, this PR will be closed by 2/18 unless updated with further comment or conflict resolution.

**Pull request closed without merging**